src/main/kotlin/no/nav/familie/ef/sak/økonomi/dto/TilkjentYtelseRestDTO.kt

```diff
@@ -0,0 +1,81 @@
+package no.nav.familie.ef.sak.økonomi.dto
+
+import java.lang.IllegalStateException
+import java.time.LocalDate
+
+data class TilkjentYtelseRestDTO(
+    val søker: String,
+    val saksnummer: String,
+    val andelerTilkjentYtelse: List<AndelTilkjentYtelseRestDTO>
+) {
+
+    fun valider() {
+        if(andelerTilkjentYtelse.size==0) {
+            throw IllegalStateException("Request trenger minst én andel tilkjent ytelse")
+        }
+
+        andelerTilkjentYtelse.forEach {
+            if(it.beløp < 0)
+                throw IllegalStateException("Beløp på andel tilkjent ytelse er negativt")
+            if(it.stønadTom < it.stønadFom)
+                throw IllegalStateException("Stønad til-og-med-dato (${it.stønadTom}) " +
+                                            "kan ikke være tidligere enn stønad fra-og-med-dato (${it.stønadFom})")
+        }
+    }
+}
+
+data class AndelTilkjentYtelseRestDTO(
+        val personIdentifikator : String,
+        val beløp: Int,
+        val stønadFom: LocalDate,
+        val stønadTom: LocalDate,
+        val type: YtelseType
+)
+
+fun TilkjentYtelseRestDTO.tilTilkjentYtelse(status: TilkjentYtelseStatus = TilkjentYtelseStatus.OPPRETTET) : TilkjentYtelse {
```

Comment:
Kan ikke disse like godt være deklarert innee i sine respektive klasser i stedet for å defineres som extension functions.

Reply:
Jo, det kan de. Er vel mer filosofisk om hver av klassene skal kjenne til den andre, eller om mapping mellom dem skal ligge "utenfor". Jeg heller mot det siste, uten at det betyr mye for meg.

Reply:
I søknad og mottak har vi adskilt deet helt i en egen mapper-pakke hvor vi har eksempelvis 
object SøknadMapper {
fun toDto(parametere ) {
 ...
}

fun toDomain(parametere) {
...
}
}

Reply:
Synes det er en fin løsning. Men er vel ikke praktisk store forskjellen mellom en `object SøknadMapper` og fil SøknadMapper med extension function?

Reply:
Extensions gir jo veldig elegant kode i bruken da. Det har du rett i.